lib/msf/core/post/file.rb

```diff
@@ -506,14 +506,15 @@ def rm_rf(*remote_dirs)
   # @param old_file [String] Remote file name to move
   # @param new_file [String] The new name for the remote file
   def rename_file(old_file, new_file)
+    raise "File to be renamed doesn't exists" unless exist?(old_file)
     if session.type == "meterpreter"
       return (session.fs.file.mv(old_file, new_file).result == 0)
+    elsif session.type == 'powershell'
+      cmd_exec("Rename-Item -Path #{old_file} -NewName #{new_file}")
+    elsif session.platform == 'windows'
+      !!(cmd_exec(%Q|move /y "#{old_file}" "#{new_file}" & if not errorlevel 1 echo true|) =~ /true/)
     else
-      if session.platform == 'windows'
-        cmd_exec(%Q|move /y "#{old_file}" "#{new_file}"|) =~ /moved/
-      else
-        cmd_exec(%Q|mv -f "#{old_file}" "#{new_file}"|).empty?
-      end
+      cmd_exec(%Q|mv -f "#{old_file}" "#{new_file}" && echo true|).strip == "true"
```

Comment:
Can we use random alphanumeric tokens instead of the word "true" like other spots do when a token is used? `Rex::Text.rand_text_alphanumeric` should do the trick.

Reply:
okay I will make it use that, also will look at other methods which are using true and not random generated alphanumeric strings and change them too.